### PR TITLE
added migration script to containerd

### PR DESCRIPTION
**Pull request by Pavan-Gunda**
**What this PR does / why we need it**:
adds containerd migration script
**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Checklist:**

- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
  - [ ] is completely transparent, will not impact the workload in any way.
  - [ ] requires running a migration script.
  - [ ] requires draining and/or replacing nodes.
  - [ ] will break the cluster.
        I.e. full cluster migration is required.

### CODE REVIEW

File: migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    - name: install pigz
+      apt:
+        name: pigz
+        state: present
+
+    - name: run kubespray on and limit to just one node.
+      become: no
+      command: pwd #ansible-playbook -i {{inventory_file}} ../../kubespray/cluster.yml --limit={{hostname.stdout}} -b
+      delegate_to: localhost
```
Comment (Xartos) [inline migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml:40]: I guess this is some leftovers?

File: migration/v2.18.0-ck8s1-v2.19.x-ck8s1/migrate-to-containerd.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    - name: install pigz
+      apt:
+        name: pigz
+        state: present
+
+    - name: run kubespray on and limit to just one node.
+      become: no
+      command: pwd #ansible-playbook -i {{inventory_file}} ../../kubespray/cluster.yml --limit={{hostname.stdout}} -b
+      delegate_to: localhost
```
Comment (Xartos) [inline migration/v2.18.0-ck8s1-v2.19.x-ck8s1/migrate-to-containerd.yml:40]: Same here

File: migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml (hunk 1 above)
Comment (Pavan-Gunda) [reply migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml]: changed it!!

File: migration/v2.18.0-ck8s1-v2.19.x-ck8s1/migrate-to-containerd.yml (hunk 2 above)
Comment (Pavan-Gunda) [reply migration/v2.18.0-ck8s1-v2.19.x-ck8s1/migrate-to-containerd.yml]: changed it

**Review by Xartos — APPROVED** [review]
LGTM! Great stuff :)

File: migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
+- hosts: k8s_cluster
+  serial: 1
+  tasks:
+
+    - name: Check that ansible version is >= 2.10.0
```
Comment (viktor-f) [inline migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml:5]: Could we also have a check to verify that `container_manager: containerd` is set?

File: migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    - name: remove docker and its dependencies
+      apt:
+        pkg:
+        - docker-ce
+        - docker-ce-cli
+        - docker-ce-rootless-extras
+        - containerd.io
+        state: absent
+        force: yes
```
Comment (viktor-f) [inline migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml:29]: According to [this guide](https://kubespray.io/#/docs/upgrades/migrate_docker2containerd) you should stop the kubelet and docker between draining the node and removing docker.

```bash
service kubelet stop
service docker stop
```

File: migration/v2.18.0-ck8s1-v2.19.x-ck8s1/upgrade-cluster.md [hunk 5]
```diff
@@ -15,3 +15,11 @@
...
 
 1. Upgrade your cluster by running `./bin/ck8s-kubespray run-playbook wc upgrade-cluster.yml -b`.
+
+1. After the upgrade, if you want to start using containerd follow [this guide](https://kubespray.io/#/docs/upgrades/migrate_docker2containerd) to migrate or set `container_manager: containerd` to `${CK8S_CONFIG_PATH}/{wc,sc}-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml` and run
+
+```
+./bin/ck8s-kubespray run-playbook {sc|wc} ../migration/v2.18.0-ck8s1-v2.19.x-ck8s1/migrate-to-containerd.yml -b
+```
+
+**NOTE**: If you are running [Compliant Kubernetes Apps](https://github.com/elastisys/compliantkubernetes-apps). Make sure you're running a version that has support for conatinerd.
```
Comment (viktor-f) [inline migration/v2.18.0-ck8s1-v2.19.x-ck8s1/upgrade-cluster.md:25]: Do we need to have this for the 2.19 upgrade as well? (I guess it does not hurt to have it here, but I had assumed that we will upgrade all clusters while on 2.18).

File: migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml (hunk 4 above)
Comment (Pavan-Gunda) [reply migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml]: Done!!

File: migration/v2.18.0-ck8s1-v2.19.x-ck8s1/upgrade-cluster.md (hunk 5 above)
Comment (viktor-f) [reply migration/v2.18.0-ck8s1-v2.19.x-ck8s1/upgrade-cluster.md:25]: Never mind, let's have this in 2.19 as well.

File: migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml (hunk 3 above)
Comment (Pavan-Gunda) [reply migration/v2.17.x-ck8s1-v2.18.x-ck8s1/migrate-to-containerd.yml]: This now has a check to see if container manager is set to contained and also sees if containerd service exists and only runs if containerd service is not running.

File: migration/v2.18.0-ck8s1-v2.19.x-ck8s1/upgrade-cluster.md (hunk 5 above)
Comment (Pavan-Gunda) [reply migration/v2.18.0-ck8s1-v2.19.x-ck8s1/upgrade-cluster.md:25]: I guess we can have it in both, as it doesn't hurt.

**Review by viktor-f — APPROVED** [review]
nit: the pre-commit change should probably be in a separate commit, but you don't have to fix that.

**Pull request merged**